src/SOFe/AwaitStd/AwaitStd.php

```diff
@@ -130,11 +130,12 @@ private function registerListener(string $event, int $priority, bool $ignoreCanc
 		$listener = new AwaitExecutor($toPlayer);
 		$this->plugin->getServer()->getPluginManager()->registerEvent(
 			$event,
-			new DummyListener,
+			static function(event\Event $event) use($listener) : void{
+				$listener->execute($event);
+			},
 			$priority,
-			$listener,
 			$this->plugin,
-			$ignoreCancelled
+			!$ignoreCancelled
```

Comment:
Maybe refactor the parameter as well so that it's less confusing?

Reply:
I did that originally, but decided to revert it since it's BC breaking. I'll do it though.